src/app/g3w-ol/controls/interactioncontrol.js

```diff
@@ -252,11 +366,72 @@ proto.setLayers = function(layers=[]){
 };
 
 /**
- * called when project change
- * @param layers
+ * @param { 'intersects' | 'within' } spatialMethod
+ * 
+ * @listens change-spatial-method
+ * 
+ * @since 3.8.0
+ */
+proto.handleChangeSpatialMethod = function(spatialMethod) {
+  let eventKey = null;
+
+  const unlistenSpatialMethodChange = () => {
+    ol.Observable.unByKey(eventKey);
+    eventKey = null;
+  };
+
+  this.on('toggled', ({toggled}) => {
+    if (true === toggled) {
+      eventKey = this.on('change-spatial-method', this.runSpatialQuery);
+    } else if (null !== eventKey) {
+      unlistenSpatialMethodChange();
+      // reset to default
+      this.setSpatialMethod(spatialMethod);
+      this.clear();
+    }
+  })
+};
+
+/**
+ * Handle temporary layers added by `addlayers` map control (Polygon or Multipolygon)
+ *
+ * @listens CatalogService~addExternalLayer
+ * @listens CatalogService~removeExternalLayer
+ *
+ * @since 3.8.0
  */
-proto.change = function(layers=[]){
-  //to owerwite to each control
+proto._handleExternalLayers = function() {
+  const CatalogService = GUI.getService('catalog');
+
+  // 0. store unwatches of extenal layers (selected or visible)
+  const unWatches = {};
+
+  // 1. update list `this.externalLayers`
+  // 2. update list `unWatches` of layer un-watchers (based on unique name of layer)
+  // 3. call `this.handleAddExternalLayer`
+  CatalogService.onafter('addExternalLayer', ({layer, type}) => {
+    if ('vector' === type) {
+      this.externalLayers.push(layer);
+      unWatches[layer.name] = [];
+      this.handleAddExternalLayer(layer, unWatches);
+    }
+  });
+
+  // 4. clean up any previously attached event listener
+  CatalogService.onafter('removeExternalLayer', ({name, type}) => {
+    if ('vector' === type) {
+      this.externalLayers = this.externalLayers.filter(layer => {
+        if (name === layer.name) {
+          this.handleRemoveExternalLayer(layer);
+          (layer === this.selectedLayer) && this.setSelectedLayer(null);
+        }
+        return name !== layer.name;
+      });
+      unWatches[name].forEach(unWatch => unWatch());
+      delete unWatches[name];
+    }
+  });
+
 };
```

Comment:
This function seems to be in use only within the `BaseQueryPolygonControl` subclass

Ref: https://github.com/g3w-suite/g3w-client/pull/341#discussion_r1129147291